lib/core/unittest/CLoopProgressTest.cc

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+#include "CLoopProgressTest.h"
+
+#include <core/CLogger.h>
+#include <core/CLoopProgress.h>
+
+#include <test/CRandomNumbers.h>
+
+using namespace ml;
+
+using TSizeVec = std::vector<std::size_t>;
+
+void CLoopProgressTest::testShort() {
```

Comment:
I think it'd be be nice to add a simple test (say 4 steps) that also verifies the exact progress in after each increment.

Reply:
See below.